### PR TITLE
Formrecognizer update to ResponseError usage

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/administration/models/ModelOperationDetails.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 
 import com.azure.ai.formrecognizer.implementation.util.ModelOperationDetailsHelper;
+import com.azure.core.models.ResponseError;
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/administration/models/ModelOperationDetails.java:8]: @alzimmermsft/ @srnagar Do we know why we do not allow setting properties of `details`, `innerError` or `target` on the [ResponseError](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/models/ResponseError.java) model in core?

Comment (srnagar) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/administration/models/ModelOperationDetails.java:8]: We had an [arch board discussion](https://github.com/Azure/azure-sdk/issues/3221) on this topic. The gist of the discussion is that only the message and error code is considered API contract. All other details like inner error, target etc. are not part of the API contract and can change in ways that can break backward compatibility. So, the response error type in azure-core only exposes message and code. 

The recording is in the issue if you are interested in the full discussion.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/util/Transforms.java [hunk 2]
```diff
@@ -270,6 +267,14 @@ public static Throwable mapToHttpResponseExceptionIfExists(Throwable throwable)
         return throwable;
     }
 
+    public static HttpResponseException mapResponseErrorToHttpResponseException(Error error) {
+        return new HttpResponseException(
+            error.getMessage(),
+            null,
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/util/Transforms.java:273]: This should not be `null`. We should include the HttpResponse here as that contains useful details like HttpRequest, the response headers and the HTTP status code.

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/util/Transforms.java:273]: This is a client-side error being thrown as an exception and we do not have a way to populate HttpResponse here.

**Review by mssfang — APPROVED** [review]
Looks good!

**Pull request merged**